collector/utils.py

```diff
@@ -0,0 +1,52 @@
+import logging
+from threading import current_thread
+
+import requests
+
+from .env import SSL_VERIFY
+
+CHUNK = 10240
+MAX_RETRIES = 3
+LOGGER = logging.getLogger()
+
+# pylama:ignore=E1101
+requests.packages.urllib3.disable_warnings()
+
+
+class RetryFailedError(requests.HTTPError):
+    """Exception raised when the HTTP retry fails.
+
+    Behaves as requests.HTTPError.
+    """
+
+
+def retryable(method: str, *args, **kwargs) -> requests.Response:
+    """Retryable HTTP request.
+
+    Invoke a "method" on "requests.session" with retry logic.
+    :param method: "get", "post" etc.
+    :param *args: Args for requests (first should be an URL, etc.)
+    :param **kwargs: Kwargs for requests
+    :return: Response object
+    :raises: HTTPError when all requests fail
+    """
+    thread = current_thread()
+    request_kwargs = dict(verify=SSL_VERIFY)
```

Comment:
Looks like for testing this locally, we need to set `verify=false`
But does work well in OpenShift env, with the value set to `True`

Reply:
Yes. Basically, what I did [here](https://github.com/ManageIQ/aiops-data-collector/pull/33/files#diff-02fc8c20454cd8f20ec0bc299ee2ea00R9) is that you can have a `SSL_VERIFY` variable set in your environment. And if it's anything else than True (even when it's not set), it defaults to `verify=True`. So locally you can set your env to disable SSL verification, but in deployment it's enforced.